### PR TITLE
Refactor App lifecycle

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR refactors the App lifecycle by not creating a new App instance each request.  One App is created at server creation time, and passed to any services needing it.

To avoid cyclical dependencies, the few server methods needed by App are provided via interface.

#### Ticket Link
https://focalboard-community.octo.mattermost.com/workspace/qgsck6cts3fwpqwyjiupjm5cde?id=47aa9bb4-6967-4a96-83c7-11bd6b20f1eb&v=ca1a5441-4c1c-4d0e-856e-88cc744576c8&c=8b6fc880-163b-4f70-8cfa-2be198b487be

### CODE REVIEW

File: server/app/app.go [hunk 1]
```diff
@@ -25,14 +30,14 @@ type App struct {
...
-	wsServer     *ws.Server
+	wsServer     WebsocketServer
 	filesBackend filestore.FileBackend
 	webhook      *webhook.Client
 	metrics      *metrics.Metrics
 	logger       *mlog.Logger
 }
 
-func New(config *config.Configuration, wsServer *ws.Server, services Services) *App {
+func New(config *config.Configuration, wsServer WebsocketServer, services Services) *App {
```
Comment (jespino) [inline server/app/app.go:40]: Is there any reason to not follow the proposed approach of `type App struct {services Services}`, also is there any reason to pass the `wsServer` outside and inside the `services` param? (I know that are not related to your changes).

Comment (wiggin77) [reply server/app/app.go:40]: We end up with calls like `a.services.logger.Debug(...)` which gets tedious. I think we still achieve the abstraction/separation even though the App has each service as a separate field.

The only reason the wsServer` is not passed in the services struct is because `ws.Server` is not in the services package.

Comment (jespino) [reply server/app/app.go:40]: ok, makes sense.

**Review by jespino — APPROVED** [review]
LGTM

**Review by mgdelacroix — APPROVED** [review]
Awesome!
